tesseract-clickhouse/src/sql/primary_agg.rs

```diff
@@ -207,6 +229,17 @@ pub fn primary_agg(
             sub_queries,
             dim_subquery.foreign_key
         );
+
+        // Wrap with rate subquery if there is a rate calculation
```

Comment:
If rate is going to wrap the primary aggregation, would it be possible to spin it into its own module? I had wondered about this since you were also having to modify rca.

For example, growth is a separate module, which wraps primary agg or rca.